### ISSUE TITLE
Training parameters for FFDNet

### ISSUE THREAD

**Issue by betty97** [cszn/KAIR#5]
Hi,
Thanks for the code. I want to train the FFDNet, however I don't understand why the training parameters in the JSON file are different from those in the paper: batch normalization is not activated (and therefore, there is no merging neither), the patch size is set to 64 instead of 70 (or 50 for color image denoising), the batch size is set to 64 instead of 128,  L1 is set as the loss function instead of the one shown in the paper, and the generation of training patches is also different since the notion of epoch is not implemented. Could you explain the reason why these settings are changed, especially the absence of batch normalization? Thank you very much!

**Comment by cszn** [comment]
- L1 loss is widely-used in recent image restoration papers.
- Compared to DnCNN, FFDNet considers a wide range of noise levels and directly predict the clean image, thus, BN is less effective for FFDNet. I think removing BN and using L1 loss can also get the similar results although sometimes BN can stablize the training.
- Do not worry about the patch size and batch size, they are not important for the final performance if they are in a proper range.
- Anyway, for your own work, you do not need to follow the parameter setting in the paper.

**Comment by betty97** [comment]
Great, thank you!

**Comment by tongchangD** [comment]
> * L1 loss is widely-used in recent image restoration papers.
> * Compared to DnCNN, FFDNet considers a wide range of noise levels and directly predict the clean image, thus, BN is less effective for FFDNet. I think removing BN and using L1 loss can also get the similar results although sometimes BN can stablize the training.
> * Do not worry about the patch size and batch size, they are not important for the final performance if they are in a proper range.
> * Anyway, for your own work, you do not need to follow the parameter setting in the paper.

请问这个patch size 设置为多少是比较合适的，我现在手上的一批图尺寸太小了，设置LR的patch是32，感觉比较糊，肉眼基本都看不出什么东西
